redbadger/continous-compliance
# Store compressed compliance folder into a bucket

Closes #20 

## How to test

1. Go to `count-dracula` code and pull recent remote branches
```sh
git fetch
```
2. Go to branch `test-continous-compliance-v05`
```sh
git checkout test-continous-compliance-v05
```
3. Do an empty commit and push to remote
```sh
git commit --allow-empty -m "Trigger continous-compliance CI/CD"
git push
```
4. Go to count dracula [actions](https://github.com/redbadger/count-dracula/actions)
5. Wait until `continous compliance` step on **continous-compliance** job finished and open the job.
6. You should see a message like this
```txt
Compliance folder compressed
Decoding gcp-application-credentials 👀
Authenticating with Google Cloud storage ✍🏻
Uploading zip file to Google Cloud storage 📡
Compliance evidence uploaded to https://storage.googleapis.com/count-dracula-continous-compliance-prod/1618319898.zip?GoogleAccessId=terrafrom-ci%40count-dracula-production.iam.gserviceaccount.com&Expires=16447017600&Signature=NlkZNxi6cuDMwEpDEMNCxTuDC8GHIpXqw8VWx52PwrtGu0TbbRg0IV7NC%2F5zNTMp2%2BqkswGxvkfOIwocat0glSVIX3y4h%2B6zam%2B0EGmBKWRcd7xE5FO0g06SRlM%2B02uKK58bfkM2LqrVMe5YbZLpYUoqYJXYGLaCvWC0l8f5plJx4Dc%2FyEGqOhUTej0HT5S%2FuJIg8kjKm4dSyAjQHgwTNr%2BpH9xUpbPWv3Hae6OtVN8o0qt05VCWfbTBKyxLsfDVCQU%2FCtI4QN5CAIQUA6aOfhQs4Sm7h%2FcQteq4Cl9IhvRggpuTT1tJZR2elsaYOuXa6RPybxh8buca5CfL7AC3LA%3D%3D
```
7. Click the URL on the last line and should trigger a file download
8. 🚀 
![](https://media.giphy.com/media/xX8jOgoBnVrY4/giphy.gif)
